packages/at_client/lib/src/client/at_client_impl.dart

```diff
@@ -907,6 +910,48 @@ class AtClientImpl implements AtClient, AtSignChangeListener {
     }
   }
 
+  @override
+  Future<AtResponse> setSPP(String spp) async {
+    // SPP should be 6 characters PIN. Throw exception if its less
+    // or more than 6 characters
+    if (spp.length != 6) {
+      throw InvalidPinException.message("$spp should be 6 characters");
+    }
+    // Validate the SPP. The SPP should contain only alpha-numeric characters.
+    // Any special characters or any characters other than aplha-numeric characters
+    // are not allowed. Throw an exception
+    bool hasMatch = RegExp(r'[\W-]+').hasMatch(spp);
+    if (hasMatch) {
+      throw InvalidPinException.message("$spp is not a valid SPP");
+    }
+    String? otpVerbResponse;
+    try {
+      otpVerbResponse =
+          await _remoteSecondary?.executeCommand('otp:put:$spp\n', auth: true);
+    } on AtLookUpException catch (e) {
+      throw InvalidPinException(e.errorCode, e.errorMessage);
```

Comment:
we have to check the error code and then throw InvalidPinException. It can be any AtLookUpException here